### PR TITLE
feat(carousel): use uib- prefix

### PR DESCRIPTION
**Pull request by BobbieBarker**
Added uib prefixing to the carousel directive.

### CODE REVIEW

**Comment by wesleycho** [comment]
Clean up the history (there should not be a merge commit) - also we need tests for the deprecation warning.

**Comment by BobbieBarker** [comment]
i knew I was forgetting something!

**Comment by Foxandxss** [comment]
There is a list of things that needs to be fixed:

Slide needs prefixing
You are creating the carousel module twice
Remove the controller from the deprecated part
Remove the animation from the deprecated part
Make the deprecated carousel use the prefixed controller
Fix the deprecation comment to use carousel
Change $alertSupressWarning to collapse (in all different places)
Add tests for deprecation warning
Fix all the tests to use the new prefixed versions

Perhaps I am forgetting something, but that is a good start.

**Comment by Foxandxss** [comment]
Nice werk. Remaining.

Remove all the extra docs on the deprecated part (I want to remove them all but that is a different topic).
The deprecated slide needs deprecation messages too
You still need to change the $alertSuppress thingy to carousel.

**Comment by Foxandxss** [comment]
Nice.

So, the demo is still using non prefixed slide
For the deprecation tests, no need of two describes. Check how the others are done. One describe with one tests with logs suppressed and one with all errors.

File: src/carousel/carousel.js [hunk 1]
```diff
@@ -413,3 +405,65 @@ function ($injector, $animate) {
     }
   };
 }]);
+
```
Comment (wesleycho) [inline src/carousel/carousel.js]: Remove 3 of the lines

File: src/carousel/carousel.js [hunk 2]
```diff
@@ -413,3 +405,65 @@ function ($injector, $animate) {
...
+
+
+
+/* Deprecated alert below */ 
+
+/**
```
Comment (wesleycho) [inline src/carousel/carousel.js]: No need for this ngdoc, especially of a deprecated component.

File: src/carousel/test/carousel.spec.js [hunk 3]
```diff
@@ -67,12 +67,12 @@ describe('carousel', function() {
...
-        '<carousel interval="interval" no-transition="true" no-pause="nopause">' +
-          '<slide template-url="foo/bar.html"></slide>' + 
-        '</carousel>'
+        '<uib-carousel interval="interval" no-transition="true" no-pause="nopause">' +
+          '<uib-slide template-url="foo/bar.html"></uib-slide>' + 
+        '</uib-carousel>'
       )(scope);
       $rootScope.$digest();
-
+
```
Comment (wesleycho) [inline src/carousel/test/carousel.spec.js]: Remove the whitespace

**Pull request closed without merging**